cli/src/meta/mod.rs

```diff
@@ -52,32 +69,63 @@ pub enum ContentEncoding {
 }
 
 impl ContentEncoding {
-    pub fn encode(&self, data: Vec<u8>) -> anyhow::Result<Vec<u8>> {
+    /// encode the data based on the variant
+    pub fn encode(&self, data: &Vec<u8>) -> anyhow::Result<Vec<u8>> {
+        Ok(match self {
+            ContentEncoding::None | ContentEncoding::Identity => data.clone(),
+            ContentEncoding::Deflate => deflate::deflate_bytes_zlib(data),
+        })
+    }
+
+    /// decode the data based on the variant
+    pub fn decode(&self, data: &Vec<u8>) -> anyhow::Result<Vec<u8>> {
         Ok(match self {
-            ContentEncoding::None | ContentEncoding::Identity => data,
-            ContentEncoding::Deflate => deflate::deflate_bytes(&data),
+            ContentEncoding::None | ContentEncoding::Identity => data.clone(),
+            ContentEncoding::Deflate => match inflate::inflate_bytes_zlib(data) {
+                Ok(v) => v,
+                Err(error) => Err(anyhow::anyhow!(error))?
+            },
         })
     }
 }
 
-#[derive(serde::Serialize, Copy, Clone, EnumString, EnumIter, strum::Display, Debug, PartialEq)]
+#[derive(serde::Serialize, Copy, Clone, EnumString, EnumIter, strum::Display, Debug, PartialEq, serde::Deserialize)]
 #[serde(rename_all = "kebab-case")]
 #[strum(serialize_all = "kebab-case")]
 pub enum ContentLanguage {
     None,
     En,
 }
 
-#[derive(PartialEq, Debug)]
-pub struct RainMetaDocumentV1Item {
+/// # Rain Meta Map
+/// represents a rain meta data and configuration that can be cbor encoded or unpacked back to the meta types
+#[derive(PartialEq, Debug, Clone)]
+pub struct MetaMap {
```

Comment:
why was this renamed? "meta map" means nothing to me, the original name was descriptive and versioned

if we have a v2 of rain document that also has maps, this "meta map" name will cause problems

Reply:
well the reason I changed it because RaimMetaDocument explicitly refers to a meta seq with Raindocument magic number, but this is just a single meta map, but feel free to update it as you see fit

Reply:
@rouzwelt but the original struct was rain meta document _item_, as in a single item in the seq

Reply:
yea so having that prefix + "item" sounds a bit vague to me, but it's not deal breaker, I just personally thought this name sounds better, but if you think the former was better, let me know

Reply:
@rouzwelt i'd like to use the original name, or something equally specific and versioned